middleware/is-auth.js

```diff
@@ -0,0 +1,25 @@
+const jwt = require('jsonwebtoken');
+
+module.exports = (req, res, next) => {
+  const authHeader = req.get('Authorization');
+  if (!authHeader) {
+    const error = new Error('Not authenticated.');
+    error.statusCode = 401;
+    throw error;
+  }
+  const token = authHeader.split(' ')[1];
+  let decodedToken;
+  try {
+    decodedToken = jwt.verify(token, 'somereallylongandunpredictabletext');
```

Comment:
never ever highlight sensitive information in the sources - instead you need to have a `.env` file,  use `dotenv` npm module to get `secretKey` from `.env` file, then use `process.env.SECRET_KEY` in the source code.

Please check these docs:
https://www.codegrepper.com/code-examples/javascript/dotenv+express+js
https://www.npmjs.com/package/dotenv